pkg/api/handlers/libpod/images.go

```diff
@@ -193,17 +210,29 @@ func ExportImage(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := newImage.Save(r.Context(), name, query.Format, tmpfile.Name(), []string{}, false, query.Compress); err != nil {
+	if err := newImage.Save(r.Context(), name, query.Format, output, []string{}, false, query.Compress); err != nil {
 		utils.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest, err)
 		return
 	}
-	rdr, err := os.Open(tmpfile.Name())
+	// if dir format, we need to tar it
+	if query.Format == "oci-dir" || query.Format == "docker-dir" {
+		rdr, err := utils2.Tar(output)
+		if err != nil {
+			utils.InternalServerError(w, err)
+			return
+		}
+		defer os.RemoveAll(output)
+		defer rdr.Close()
+		utils.WriteResponse(w, http.StatusOK, rdr)
+		return
+	}
+	rdr, err := os.Open(output)
 	if err != nil {
 		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "failed to read the exported tarfile"))
 		return
 	}
 	defer rdr.Close()
-	defer os.Remove(tmpfile.Name())
+	//defer os.Remove(output)
```

Comment:
Is this supposed to be commented?

Reply:
This also seems like it needs to be moved up above the block at line 217, and always be a RemoveAll